compaction/task_manager_module.cc

```diff
@@ -266,6 +266,7 @@ future<tasks::task_manager::task::progress> compaction_task_impl::get_progress(c
 }
 
 future<> major_keyspace_compaction_task_impl::run() {
+    co_await replica::database::flush_on_all_shards(_db);
```

Comment:
But let's not block this series by this comment.

Reply:
It's a good point.
I need to rebase anyway, so I'll give it a shot.

Reply:
we'd need a wrapper in database like: database::compact_all_sstables(table& t);

Reply:
@raphaelsc flushing all tables needs to happen once per major compaction while the call to compact_all_sstables is too low in the stack (top-level -> shard -> table) so you don't want to flush all tables per table.  We can do it once per shard, but it will still be in the task_manager module (shard_major_keyspace_compaction_task_impl)

Reply:
If we're flushing multiple (say hundred) keyspaces, one after the other, it could be a problem that we're producing many small sstables, right?

Reply:
for every keyspace compacted, we're flushing every memtable across all keyspaces and shards. early flushes contribute not only with additional compaction work but also write amplification.

Reply:
the write amplifications comes from that fact that you're adding a small base sstable size, meaning more size tiers for the data to travel through.  that's somehow mitigated with min_sstable_size, but it's still additional write amp.

Reply:
I tend to think this series might impose a regression on certain users.

Reply:
Users that rely on periodic majors and have many keyspaces might face a problem. But on the other hand, the more keyspaces you have, the smaller your tables are.

Reply:
That's a valid point.  https://github.com/scylladb/scylladb/pull/15971 will limit the "window of vulnerability" also for major compaction, so users that want to purge every last tombstone in the table should run repair and flush first before running major compaction, but in the common case major compaction can do best efforst with the on-going limitations of tombstone_gc (in whatever mode it is), and now commitlog considerations.